taskvine/src/worker/vine_process.c

```diff
@@ -55,15 +55,26 @@ extern char * workspace;
 extern struct list *library_list;
 extern struct hash_table *library_ids;
 
-static int create_sandbox_dir( struct vine_process *p )
+static int create_sandbox_dir( struct vine_process *p, int mini_task)
 {
 	p->cache_dir = string_format("%s/cache",workspace);
-  	p->sandbox = string_format("%s/t.%d", workspace,p->task->task_id);
+
```

Comment:
Don't duplicate the `string_format` calls here.
Instead do something like this:

```
const char *type = mini_task ? "m" : "t";
```

Then just use those strings when calling string_format.